### PR TITLE
Cookie banner

### PR DESCRIPTION
**Pull request by nicl**
Adds cookie banner.

screen shot 2018-10-10 at 10 46 37

image

Other breakpoints:

screen shot 2018-10-10 at 16 44 58
screen shot 2018-10-10 at 16 45 06

### CODE REVIEW

File: frontend/components/CookieBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import { palette } from '@guardian/pasteup/palette';
+import { sans, serif } from '@guardian/pasteup/fonts';
+import { Container } from '@guardian/guui';
+import TickIcon from '@guardian/pasteup/icons/tick.svg';
+import { getCookie, addCookie } from '../lib/cookie';
+
+const banner = css`
+    position: fixed;
+    bottom: 0;
+    background-color: ${palette.sport.dark};
```
Comment (AWare) [inline frontend/components/CookieBanner.tsx]: Use cases like this makes me regret calling it `sport` in the palette.

**Comment by GHaberis** [comment]
@nicl do you have any screenshots of how this looks at a mobile and tablet breakpoints?

File: frontend/components/CookieBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+export default class CookieBanner extends Component<{}, { show: boolean }> {
+    constructor(props: {}) {
+        super(props);
+
+        this.state = {
+            show: false,
+        };
+
+        this.accept = this.accept.bind(this);
```
Comment (AWare) [inline frontend/components/CookieBanner.tsx]: do typescript public methods need binding?

File: frontend/components/CookieBanner.tsx (hunk 1 above)
Comment (nicl) [reply frontend/components/CookieBanner.tsx]: ha true

File: frontend/components/CookieBanner.tsx (hunk 2 above)
Comment (AWare) [reply frontend/components/CookieBanner.tsx]: https://basarat.gitbooks.io/typescript/docs/tips/bind.html and https://stackoverflow.com/questions/20627138/typescript-this-scoping-issue-when-called-in-jquery-callback
seem to suggest `public accept = () =>` would be better

Comment (nicl) [reply frontend/components/CookieBanner.tsx]: I'm pretty sure it didn't work without the binding but will check.

**Comment by nicl** [comment]
@GHaberis added some additional screenshots to description. To be honest, there isn't really any responsive behaviour here other than being in our responsive container component.

File: frontend/components/CookieBanner.tsx (hunk 2 above)
Comment (nicl) [reply frontend/components/CookieBanner.tsx]: Cool thanks - that does look nicer!

File: frontend/components/CookieBanner.tsx (hunk 1 above)
Comment (SiAdcock) [reply frontend/components/CookieBanner.tsx]: Gosh, I wonder if we should name these variables after the appropriate colours of the rainbow?

Comment (nicl) [reply frontend/components/CookieBanner.tsx]: But seriously, probably non-semantic names are best - e.g. like we do with the neutral colours.

Comment (GHaberis) [reply frontend/components/CookieBanner.tsx]: Hmm an argument for semantic names is that they're easier to reason about when say dynamically picking a palette colour dependent on an articles pillar. EG.

 ``
const myStyle = pillarName => css`
    color: ${palette[pillarName].pastel}
` ;
``

If we were just using numbers as we do for neutrals we'd need to be sure each pillar's palette uses the same number for the pastel colour, it seems easier to make a mistake and mismatch the colour to the wrong number than the wrong semantic name.

**Pull request merged**